src/treeview_switches.ml

```diff
@@ -240,17 +244,21 @@ let register extension =
           in
           `Promise items )
     in
-    let getTreeItem ~element = Promise.return element in
-    let event_emitter = Vscode.EventEmitter.make () in
-    let event = Vscode.EventEmitter.event event_emitter in
+    let getTreeItem ~element = `Value element in
+    let module EventEmitter =
+      Vscode.EventEmitter.Make (Interop.Js.Or_undefined (TreeItem)) in
+    let event_emitter = EventEmitter.make () in
+    let event = EventEmitter.event event_emitter in
+    let module TreeDataProvider = Vscode.TreeDataProvider.Make (Vscode.TreeItem) in
```

Comment:
This is great, once this PR is in, I'll refactor the `Treeview_switches` module to use a custom module instead of `TreeItem`. At the moment, we serialize the `Dependency.t` as a string in the ID of the `TreeItem` and get it back in `getChildren`, this is a really unwelcomed hack, so can't wait to get rid of it 🙌